docs/archetypes/authoring-guide.md

```diff
@@ -333,32 +333,110 @@ module telemetryCustomerUsageAttribution '../../azresources/telemetry/customer-u
 
-- [AzureSubscriptionGUID]\_[TemplateName].json
-- [AzureSubscriptionGUID]\_[TemplateName]\_[DeploymentLocation].json
+### Management Group Id detection
+
+Management Group Id must be unique within an Azure environment.  Since a management group hierarchy can be up to 6 levels deep with many child management groups , it is important to ensure uniqueness across all management groups ids.
+
+There are two approaches for achieving uniquness:
+
+1. Use the parent management group id as the prefix for every child management group id.  For example:
+
+    ```none
+        - pubsec
+          - pubsecLandingZones
+            - pubsecLandingZonesDevTest
+    ```
+
+    In this approach, the prefix creates the natural uniqueness.
+
+2. Create a management group hierarchy where every management group id can be unique and it is manually checked.  For example:
+
+    ```none
+        - pubsec
+          - LandingZones
+            - DevTest
+    ```
 
-The subscription GUID is needed by the pipeline; since it's not available in the file contents, it is specified in the config file name.
+    In this approach, you must ensure all management group ids are unique yourself.
 
-The template name/type is a text fragment corresponding to a path name (or part of a path name) under the '/landingzones' top level path. It indicates which Bicep templates to run on the subscription. For example, the machine learning path is `/landingzones/lz-machinelearning`, so we remove the `lz-` prefix and use `machinelearning` to specify this type of landing zone.
+The `subscription-ci` management group detection logic is built based on approach #1.  However it will also work for both scenarios.
 
-The deployment location is the short name of an Azure deployment location, which may be used to override the `deploymentRegion` YAML variable. The allowable values for this value can be determined by looking at the `Name` column output of the command: `az account list-locations -o table`.
+To support approach #1:
+
+- Folder structure in `config/subscription/` is created without including the prefixes.  For example:
+
+    ```none
+          config/subscription/CanadaESLZ-main
+            - pubsec
+                - LandingZones
+                    - DevTest
+    ```
+
+- `subscription-ci` will then take the folder structure and concatenate it to create the management group id.  In this example `DevTest` management group id will be `pubsecLandingZonesDevTest`.
+
+To support approach #2:
+
+- Folder structure in `config/subscription/` should be flat.  For example:
+
+    ```none
+          config/subscription/CanadaESLZ-main
+            - pubsec
+            - LandingZones
+            - DevTest
+    ```
+
+- `subscription-ci` will then take the folder name as the structure (since there aren't any sub folders).  In this example `DevTest` management group id will be `DevTest`.
+
+    **Advanced Customization (optional)**
```

Comment:
Remove this section entirely (Advanced Customization). The original management group id composition (based on a hugely flexible folder hierarchy) will accommodate any folder organization strategy desired. Modifying bash script at this level is very dangerous, i.e. avoid unless necessary, and it's not necessary. We do not want to be diagnosing issues related to this type of customization.